hilbert/factories.py

```diff
@@ -123,22 +187,47 @@ def build_mle_sample_solver(
         init=get_init_embs(init_embeddings_path, device),
         device=device
     )
-
-    if balanced:
-        print('CPU loader for balanced samples.')
-        loader_class = h.loader.CPUSampleLoader
+    if gibbs:
+        print("Using Gibbs sampling.")
+        loader = h.loader.GibbsSampleLoader(
+            cooccurrence_path=cooccurrence_path,
+            learner=learner,
+            temperature=temperature,
+            batch_size=batch_size,
+            device=device,
+            verbose=verbose
+        )
     else:
-        loader_class = h.loader.GPUSampleLoader
-    loader = loader_class(
-        cooccurrence_path=cooccurrence_path, 
-        temperature=temperature,
-        batch_size=batch_size,
-        device=device, 
-        verbose=verbose
-    )
+        if balanced:
+            print('CPU loader for balanced samples.')
+            loader_class = h.loader.CPUSampleLoader
+        else:
+            loader_class = h.loader.GPUSampleLoader
+
+        loader = loader_class(
+            cooccurrence_path=cooccurrence_path,
+            temperature=temperature,
+            batch_size=batch_size,
+            device=device,
+            verbose=verbose,
+            remove_threshold=remove_threshold,
+        )
+
 
     optimizer = get_optimizer(opt_str, learner, learning_rate)
 
+    if scheduler_str is not None:
+        lr_scheduler = get_lr_scheduler(scheduler_str=scheduler_str,
```

Comment:
I know that's a popular indent style, but in this code base we're always using just a four-space hanging indent style.
```suggestion
        lr_scheduler = get_lr_scheduler(
            scheduler_str=scheduler_str,
```